perllib/FixMyStreet/Script/Merton/SendWaste.pm

```diff
@@ -61,18 +61,18 @@ sub set_echo_id {
     my ($self, $problem) = @_;
 
     unless ($problem->get_extra_field_value('echo_id')) {
+        $problem->discard_changes;
         my $cobrand = $problem->get_cobrand_logged;
         my $cfg = $cobrand->feature('echo');
         my $echo = Integrations::Echo->new(%$cfg);
         my $event = $echo->GetEvent($problem->external_id);
         return unless $event && $event->{Id};
 
-        my $row2 = FixMyStreet::DB->resultset('Problem')->search({ id => $problem->id }, { for => \'UPDATE' })->single;
```

Comment:
Oh, I was happy with this, because it meant this extra change wouldn't overwrite any simultaneous extra change. I only meant adding a discard_changes after you've done it (so that $problem picks up the change you made with $row).

Reply:
Actually, SendReport::Open311 always calls discard_changes after calling open311_update_missing_data, so I'm not sure discard_changes is needed at all. The tests still pass without it - what was the issue you were looking to fix?

Reply:
going to merge without it, and if I've introduced a bug, apologies!